jsamchineme/send-it
# Ft change password 162723402

#### What does this PR do?
- Implement password reset feature
#### Description of Task to be completed?
- Add email notification on password change request
- Add email notification on password change success
- Write tests for endpoints
#### How should this be manually tested?
- pull branch and run _npm run start:dev_
- POST /api/v1/auth/reset with request body of `{email: 'jsamchineme@gmail.com', scope: 'testing'}`
The request should return a `resetToken`
- PUT /api/v1/auth/reset with request body of `{email: 'jsamchineme@gmail.com', token: {resetToken}, password: 'newpassword'}`
#### Any background context you want to provide?
- An email will be sent to the user's email address. In production, the resetToken is meant to be returned in the response body for security reasons.
- The request field scope: testing is supplied to allow the return of a resetToken which is required to change the password
#### What are the relevant pivotal tracker stories?
- https://www.pivotaltracker.com/story/show/162723402
#### Screenshots (if appropriate)
None
#### Questions:`
None
